cogs/committee_actions_tracking.py

```diff
@@ -105,8 +205,161 @@ async def _create_action(
             raise InvalidActionDescriptionError(
                 message=DUPLICATE_ACTION_MESSAGE
             ) from create_action_error
+        await self._update_action_board()
         return action
 
+    @overload
+    @staticmethod
+    async def get_user_actions(
+        action_user: discord.Member | discord.User, status: list[str]
+    ) -> list[AssignedCommitteeAction]: ...
+
+    @overload
+    @staticmethod
+    async def get_user_actions(
+        action_user: "Iterable[discord.Member]",
+        status: list[str],
+    ) -> dict[discord.Member, list[AssignedCommitteeAction]]: ...
+
+    @staticmethod
+    async def get_user_actions(
+        action_user: "discord.Member | discord.User | Iterable[discord.Member]",
+        status: str | list[str],
+    ) -> list[AssignedCommitteeAction] | dict[discord.Member, list[AssignedCommitteeAction]]:
+        """
+        Get the actions for a given user.
+
+        Takes in the user and returns a list of their actions.
+        """
+        if isinstance(action_user, (discord.User, discord.Member)):
+            user_actions: list[AssignedCommitteeAction] = [
+                action
+                async for action in AssignedCommitteeAction.objects.filter(
+                    status=status,
+                    discord_member__discord_id=int(action_user.id),
+                )
+            ]
+
+            return user_actions
+
+        actions: list[AssignedCommitteeAction] = [
+            action async for action in AssignedCommitteeAction.objects.select_related().all()
+        ]
+
+        committee_actions: dict[discord.Member, list[AssignedCommitteeAction]] = {
+            committee: [
+                action
+                for action in actions
+                if str(action.discord_member.discord_id) == str(committee.id)
+                and action.status in status
+            ]
+            for committee in action_user
+        }
+
+        return {
+            committee: actions for committee, actions in committee_actions.items() if actions
+        }
```

Comment:
```suggestion
        return {
            inner_action_user: user_actions
            for inner_action_user in action_user
            if (user_actions := cls._get_user_actions(action_user=inner_action_user, status=status)) is not None
        }
```

Where `CommitteeActionsTrackingBaseCog._get_user_actions()` is defined as:
```python
@classmethod
async def _get_user_actions(cls, action_user: discord.Member | discord.User, status: "Collection[str]") -> "Collection[AssignedCommitteeAction] | None":
    filtered_user_actions: "Queryset[AssignedCommitteeAction]" = AssignedCommitteeAction.objects.filter(discord_member__discord_id=action_user.id, status__in=status)

    if not await filtered_user_actions.aexists():
        return None

    return [action async for action in filtered_user_actions]  # NOTE: List comprehension is required because wrapping with `list()` is not supported for asynchronous Django queries.
```